ONSdigital/eq-author-app
# EAR  349 - Star favourite questionnaires 🤩

### What is the context of this PR?

[Ticket](https://collaborate2.ons.gov.uk/jira/browse/EAR-349) - [Prototype](https://overflow.io/s/HSI3MZGH?node=41b6f5b0&prototype&view=fit)

Adds the ability to star your favourite questionnaires - adds a "starred" column to the questionnaires view table which you can sort in order to gather your starred questionnaires at the top.

#### Other misc changes

Various fixes for dynamoDB compatibility so tests pass.

Adds global mock for jest SVGR so we can use `{ ReactComponent }` imports from SVGs without breaking all tests

#### Implementation details

Adds a `starredQuestionnaires` field to records in the users collection - an array of questionnaire IDs.
When a questionnaire is starred, its ID is added to the list for that user. When un-starred, it is removed.

Adds a `toggleQuestionnaireStarred` mutation which performs this process, taking `questionnaireId` as input - it stars / un-stars a questionnaire for the currently logged in user.

Adds a `starred` resolver to the questionnaire type. Client code therefore doesn't need to be aware of the implementation re: storing in the `users` table - you can just query `starred` on a questionnaire and it'll give you a `true` or `false` depending on whether the currently logged in user has starred that questionnaire.

### How to review

- Check you can ⭐️ 'em
- Check you can sort questionnaires by their starred status
- Check you can un-⭐️ 'em
- Check styling matches [prototype](https://overflow.io/s/HSI3MZGH?node=41b6f5b0&prototype&view=fit)

### Image

<img width="1148" alt="Screenshot 2021-04-01 at 11 27 40" src="https://user-images.githubusercontent.com/60441612/113281344-523ba380-92dd-11eb-8ed3-ce43465209fd.png">

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process
6. - [ ] 🎉
